bindings/tflite/java/org/tensorflow/lite/Interpreter.java

```diff
@@ -0,0 +1,436 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.tensorflow.lite;
+
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * Main driver class for the IREE Java compatibility shim. Provides model
+ * creation and inference for IREE compatible TFLite models.
+ *
+ * <p>This shim aims to mimic the functionality of the Tensorflow Lite's
+ * Interpeter.java class, however, there are a few notable features IREE dosn't
+ * support:
+ *
+ * <ul>
+ *  <li> Delegates and the NNAPI
+ *  <li> Advanced interpreter options
+ *  <li> Interrupting or canceling inference before its complete
+ *  <li> Tensor method signatures
+ *  <li> Dynamic shapes
+ *  <li> String input/output isn't supported
+ * <ul>
+ *
+ * <p>In addition, this compatibility shim only accepts {@link java.nio.Buffer}
+ * type input. Consumers with scalar and array inputs can see conversion methods
+ * below.
+ *
+ * <p>Example using the Interpreter with a model with a single input/output Tensor:
+ *
+ * <pre>{@code
+ * ByteBuffer modelByteBuffer = ... load model here ....
+ * try (Interpreter interpreter = new Interpreter(modelByteBuffer)) {
+ *   interpreter.allocateTensors();
+ *
+ *   float[] input = {1, 3};
+ *   float[] output = new float[2];
+ *
+ *   int bytesInFloat = 4;
+ *   FloatBuffer inputBuffer = ByteBuffer.allocateDirect(bytesInFloat * input.length)
+ *      .order(ByteOrder.nativeOrder())
+ *      .asFloatBuffer()
+ *   FloatBuffer outputBuffer = ByteBuffer.allocateDirect(bytesInFloat * output.length)
+ *      .order(ByteOrder.nativeOrder())
+ *      .asFloatBuffer()
+ *
+ *   inputBuffer.put(input);
+ *   interpreter.run(inputBuffer, outputBuffer);
+ *   outputBuffer.get(output);
+ *
+ *   ... process output ...
+ *
+ *   interpreter.close();
+ * }
+ * }</pre>
+ *
+ * <p>If a model takes multiple inputs, use
+ *   {@link #runForMultipleInputsOutputs(Buffer[], Map)}:
+ *
+ * <pre>{@code
+ * Buffer[] inputs = {input0, input1, ...};
+ * Map<Integer, Buffer> indexToOutput = new HashMap<>();
+ * FloatBuffer ithOutput = ... allocate (native) and populate buffer ...
+ * indexToOutput.put(i, ithOutput);
+ *
+ * ByteBuffer modelByteBuffer = ... load model here ....
+ * try (Interpreter interpreter = new Interpreter(modelByteBuffer)) {
+ *   interpreter.allocateTensors();
+ *   interpreter.runForMultipleInputsOutputs(inputs, indexToOutput);
+ *
+ *   .. process output ...
+ *
+ *   interpreter.close();
+ * }
+ * }</pre>
+ *
+ * <p>Orders of inputs and outputs are determined when converting TensorFlow
+ * model to TensorFlowLite model with Toco, as are the default shapes of the
```

Comment:
pretty sure TOCO is all caps

```suggestion
 * model to TensorFlowLite model with TOCO, as are the default shapes of the
```

though https://www.tensorflow.org/lite/api_docs/java/org/tensorflow/lite/Interpreter uses this capitalization...